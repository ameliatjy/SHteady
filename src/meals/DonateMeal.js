import React, { Component } from 'react';
import {
    StyleSheet,
    View,
    Text,
    Button,
    TouchableOpacity,
    Alert
} from 'react-native';

import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/storage';

export default class DonateMeal extends Component {

    confirmdonate = () => {
        var user = firebase.auth().currentUser;

        var matric = user.displayName
        var availcredits
        firebase.database().ref('users/'+ matric).on('value', function(snapshot) {
            availcredits = snapshot.val().mealcredit;
        })

        firebase.database().ref('users/' + matric).child('mealcredit').set(availcredits-1)

        Alert.alert(
            "Successful",
            "Meal credit donated!",
            [
                {
                    text: "Ok"
                }
            ]
        )
    }

    render() {
        return(
            <View style={styles.container}>
                <Text>Are you sure you'd like to donate your meal?</Text>
                <TouchableOpacity onPress={this.confirmdonate}>
                    <Text>Yes</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
  });