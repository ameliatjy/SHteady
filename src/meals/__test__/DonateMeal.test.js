import React from 'react';
import renderer from 'react-test-renderer'

import 'firebase/database'
import DonateMeal from '../DonateMeal';

const firebase = require('firebase');
//const admin = require('firebase-admin');
// Initialize Firebase
const firebaseConfig = {
    apiKey: "AIzaSyB1oyaDAneBvtqpJJqYN_o13jWDExpRDq0",
    authDomain: "shteady-b81ed.firebaseapp.com",
    databaseURL: "https://shteady-b81ed.firebaseio.com",
    projectId: "shteady-b81ed",
    storageBucket: "gs://shteady-b81ed.appspot.com",
    messagingSenderId: "749591564782",
    appId: "1:749591564782:web:73f597ecbcf1edd21dfeff",
    measurementId: "G-CBS54V1147"
};
firebase.initializeApp(firebaseConfig);

jest.useFakeTimers();

test('Donate Meals page renders snapshot as expected', () => {
    const tree = renderer.create(<DonateMeal/>).toJSON();
    expect(tree).toMatchSnapshot();
});