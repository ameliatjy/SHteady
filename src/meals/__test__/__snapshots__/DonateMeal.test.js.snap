// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Are you sure you'd like to donate your meal?
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Yes
    </Text>
  </View>
</View>
`;
